api-service/src/controllers/CreateQueryTemplate/CreateTemplateController.ts

```diff
@@ -0,0 +1,75 @@
+        }
+        const templateName: string = req.params?.templateName;
+        const templateId: string = slug(templateName, '-');
+        const requestBody = req.body;
+        const isValidSchema = schemaValidation(requestBody, validationSchema)
+
+        if (!isValidSchema?.isValid) {
+            logger.error({ apiId, message: isValidSchema?.message, code: "QUERY_TEMPLATE_INVALID_INPUT" })
+            return ResponseHandler.errorResponse({ message: isValidSchema?.message, statusCode: 400, errCode: "BAD_REQUEST", code: "QUERY_TEMPLATE_INVALID_INPUT" }, req, res);
+        }
+
+        const isTemplateExists = await getQueryTemplate(templateId)
+        if (isTemplateExists !== null) {
+            logger.error({ apiId, message: `Template ${templateName} already exists`, code: "QUERY_TEMPLATE_ALREADY_EXISTS" })
+            return ResponseHandler.errorResponse({ message: `Template ${templateName} already exists`, statusCode: 409, errCode: "CONFLICT", code: "QUERY_TEMPLATE_ALREADY_EXISTS" }, req, res);
+        }
+
+        const { validTemplate } = await validateTemplate(requestBody);
+        if (!validTemplate) {
+            logger.error({ apiId, message: `Invalid template provided, A template should consist of variables ["DATASET", "STARTDATE", "ENDDATE"] and type of json,sql`, code: "QUERY_TEMPLATE_INVALID" })
+            return ResponseHandler.errorResponse({ statusCode: 400, message: `Invalid template provided, A template should consist of variables ["DATASET", "STARTDATE", "ENDDATE"] and type of json,sql`, errCode: "BAD_REQUEST", code: "QUERY_TEMPLATE_INVALID" }, req, res)
+        }
+
+        const data = transformRequest(requestBody, templateName);
+        await QueryTemplate.create(data)
+        logger.info({ apiId, message: `Query template created successfully` })
+        return ResponseHandler.successResponse(req, res, { status: 200, data: { template_id: templateId, template_name: templateName, message: `The query template has been saved successfully` } });
+    }
+    catch (error) {
+        logger.error(error)
+        let errorMessage: any = error;
+        const statusCode = _.get(error, "statusCode")
+        if (!statusCode || statusCode == 500) {
+            errorMessage = { code: "QUERY_TEMPLATE_CREATION_FAILED", message: "Failed to query template" }
+        }
+        ResponseHandler.errorResponse(errorMessage, req, res);
+    }
+}
+
+const transformRequest = (req: any, templateName: string) => {
+    console.log({ templateName })
+    const type: any = _.get(req, "request.query_type");
+    const query = _.get(req, "request.query")
+    const data = {
+        template_id: slug(templateName, '-'),
+        template_name: templateName,
+        query_type: type,
+        query: query
+    }
+    return data
+}
+
+const validateName = (name: string) => {
```

Comment:
name should be part of request body so this should go inside the validation schema

Reply:
done